apps/dashboard/src/components/workflow-editor/steps/step-provider.tsx

```diff
@@ -27,7 +29,25 @@ export const StepProvider = ({ children }: { children: ReactNode }) => {
     stepSlug,
   });
 
-  const value = useMemo(() => ({ isPending, step, refetch }), [isPending, step, refetch]);
+  /**
+   * We need to get the issues from the workflow response
+   * because the step is not refetched when workflow is updated
+   *
+   * TODO:
+   * 1. add all step data to workflow response
+   * 2. remove StepProvider and keep just the WorkflowProvider with step value
+   */
+  const issues = useMemo(() => {
+    const newIssues = workflow?.steps.find(
+      (s) =>
+        getEncodedId({ slug: s.slug, divider: STEP_DIVIDER }) ===
+        getEncodedId({ slug: stepSlug, divider: STEP_DIVIDER })
+    )?.issues;
+
+    return { ...newIssues };
+  }, [workflow, stepSlug]);
```

Comment:
This is a hack that was present in code before - the root cause for this is that we update steps with PUT workflow endpoint using `WorkflowProvider` and we don't refetch steps (from GET step endpoint) afterwards.

The proper fix is described in the comment here.